### PR TITLE
Add beam info onto the header, if it is defined

### PR DESCRIPTION
**Pull request by e-koch**
Use [Beam.attach_to_header](https://github.com/radio-astro-tools/radio_beam/pull/18) to add the beam information to ```SpectralCube.header``` -- @keflavich 

Missing tests.

### CODE REVIEW

File: spectral_cube/spectral_cube.py [hunk 1]
```diff
@@ -1982,8 +1982,8 @@ def to_glue(self, name=None, glue_app=None, dataset=None, start_gui=True):
                     return self._glue_app
 
             glue_app.add_datasets(self._glue_app.data_collection, result)
-        
-
```
Comment (keflavich) [inline spectral_cube/spectral_cube.py]: I think my syntax checker misses these... don't know why.

**Comment by keflavich** [comment]
:+1: and :+1: to tests too!

**Comment by keflavich** [comment]
....why does coveralls think the coverage has increased?

**Comment by astrofrog** [comment]
¯\_(ツ)_/¯

**Comment by keflavich** [comment]
your emoticon is not wearing coveralls.

**Comment by e-koch** [comment]
@keflavich - Why is a ```Beam``` object only setup when the units are Jy/beam?

**Comment by keflavich** [comment]
@e-koch good question.  It is only strictly necessary when units are Jy/beam, but it would be nice to have if it's present.  It can be slow, though - CASA FITS headers are 100's of lines of history that can take a human-noticeable amount of time to parse.  Still, maybe we should always try to read a beam?

**Comment by e-koch** [comment]
@keflavich - How about a ```read_beam=True``` like argument? And perhaps it should be disabled by default with a helpful error message for ```spectral_cube.beam``` ?

**Comment by keflavich** [comment]
Actually, I'm ok with it being on by default, but I like the `read_beam` boolean flag.  We need to have it both in the `SpectralCube` init (since that's where beam parsing happens now) and in the `read` method in `io`.

**Comment by e-koch** [comment]
@keflavich @astrofrog - Ready for a review!

**Comment by keflavich** [comment]
:+1: and looks like [tests ran](https://travis-ci.org/radio-astro-tools/spectral-cube/jobs/76150321), so I'll merge

**Comment by jpinedaf** [comment]
There is a bug generated by this implementation in the `write` function, it makes `BMIN=BMAJ`

```
In [39]: from spectral_cube import SpectralCube
In [40]: cube=SpectralCube.read('L1451mm/L1451MM-11_cvel_clean_rob05.fits')
In [41]: cube.beam
Out[41]: Beam: BMAJ=3.08794236183 arcsec BMIN=2.61326932907 arcsec BPA=-19.7857666016 deg
In [42]: cube.write('test.fits')
In [44]: cube2=SpectralCube.read('test.fits')

In [45]: cube2.beam
Out[45]: Beam: BMAJ=3.08794236183 arcsec BMIN=3.08794236183 arcsec BPA=-19.7857666016 deg
```

**Pull request merged**